### PR TITLE
Add pagination and sorting to areas v1

### PR DESCRIPTION
**Pull request by tiagojsag**

### CODE REVIEW

File: app/src/serializers/area.serializer.js [hunk 1]
```diff
@@ -26,10 +26,32 @@ const areaSerializer = new JSONAPISerializer('area', {
...
-    static serialize(data) {
-        return areaSerializer.serialize(data);
+    static serialize(data, link = null) {
+        let result;
+        if (data.docs) {
+            result = areaSerializer.serialize(data.docs);
+        } else {
+            result = areaSerializer.serialize(data);
+
+        }
```
Comment (ikas) [inline app/src/serializers/area.serializer.js]: I think you shouldn't need this `if` statement in this case (in v2 serializer it is needed, however) - `result = areaSerializer.serialize(data);` should be enough

File: app/test/e2e/v1/create-area-fw.spec.js [hunk 2]
```diff
@@ -72,7 +74,8 @@ describe('V1 - Create area FW', () => {
         });
         response.body.data.attributes.should.have.property('createdAt');
         response.body.data.attributes.should.have.property('updatedAt');
-        response.body.data.attributes.updatedAt.should.equal(response.body.data.attributes.createdAt);
+        new Date(response.body.data.attributes.updatedAt).should.closeToTime(new Date(response.body.data.attributes.createdAt), 5);
+        new Date(response.body.data.attributes.updatedAt).should.closeToTime(new Date(response.body.data.attributes.createdAt), 5);
```
Comment (ikas) [inline app/test/e2e/v1/create-area-fw.spec.js]: Duplicate? :thinking:

File: app/src/services/area.service.js [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+                        }
+                        break;
+                    default:
+                        break;
+
+                }
+            } else if (param === 'usersRole') {
+                logger.debug('Params users roles');
+                query.userId = { ...query.userId || {}, $in: query[param] };
+                delete query.usersRole;
```
Comment (ikas) [inline app/src/services/area.service.js]: Do we support sorting by `userRole`? If not, then this block is not needed.

File: app/src/services/area.service.js [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+        }
+        return query;
+    }
+
+    static async getAllAreasUserIds() {
+        logger.debug(`[AreaService]: Getting the user ids of all areas`);
+        const areas = await Area.find({}, 'userId').lean();
+        const userIds = areas.map((w) => w.userId);
+        return userIds.filter((item, idx) => userIds.indexOf(item) === idx && item !== 'legacy');
+    }
```
Comment (ikas) [inline app/src/services/area.service.js]: Is this needed?

File: app/src/services/area.service.js [hunk 5]
```diff
@@ -0,0 +1,149 @@
...
+
+    static processSortParam(sort) {
+        return sort.replace(/user.role/g, 'userRole,_id').replace(/user.name/g, 'userName,_id');
+    }
```
Comment (ikas) [inline app/src/services/area.service.js]: Same here, I don't think we need this

**Review by ikas — CHANGES_REQUESTED** [review]

File: app/src/serializers/area.serializer.js (hunk 1 above)
Comment (tiagojsag) [reply app/src/serializers/area.serializer.js]: nope, tests fail

**Review by ikas — APPROVED** [review]
LGTM :+1:

**Pull request merged**